src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +832,164 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("Total number of packages removed from assessment packages list is: {0}, ".format(len(self.__assessment_packages_removed)))
+            self.composite_logger.log_debug("Packages removed from assessment packages list: {0}, ".format(self.__assessment_packages_removed))
+
+        if not len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("Total number of packages removed from installation packages list is: {0}, ".format(len(self.__installation_packages_removed)))
+            self.composite_logger.log_debug("Packages removed from installation packages list: {0}".format(self.__installation_packages_removed))
+
+        if len(self.__assessment_packages_removed) == 0 and len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("No packages truncated")
```

Comment:
Along with packages removed by truncation, can we also add other details in here, such as, the current size of status file post truncation, the size before truncation (can be grabbed from complete_status_file)?

Will help validate our status file size, if there's any changes in Guest agent or if Guest Agent still ends up slicing through our status payload.

Reply:
hmm we have log for that information in the __create_truncated_status_file(), to get those log info into the log_truncated_packages(0 we will need to create class variable and assign those values to it though. pending for final decision on this

Reply:
I did not see a log on current size of status file **post** truncation (which was asked in my first comment) in __create_truncated_status_file(). If it is covered, please add a reference to that codepoint here

Reply:
line 911, capture in the "End patches truncation

Reply:
pending